src/UndockedRegFreeWinRT/UndockedRegFreeWinRT/catalog.cpp

```diff
@@ -82,13 +83,21 @@ struct component
 
 static unordered_map<wstring, shared_ptr<component>> g_types;
 
+std::wstring_view wstr_tolower(std::wstring& s) {
+    std::transform(s.begin(), s.end(), s.begin(),
+        [](unsigned char c) { return std::tolower(c); } // correct
```

Comment:
This will break with Unicode filenames, consider using FindStringOrdinal instead. But since we're only looking at the extension, it's probably OK.

Just a little bit of fragility that you're converting the entire string, instead of just extracting the extension. If you change your function from converting the entire string, and instead, extract and `tolower` only the extension, then that's much better, since you know the extension should *only* be '.dll' or '.exe', which won't get corrupted by `tolower`.

Reply:
I don't think using the string returned substr() of path means I am passing in the entire string is it? It should still only be the extension. Yea I can remake this method into a more specific GetExtensionLowerCase that takes in the full path and returns a new string with just the extension so it doesn't get misused. 

In the next PR :)